### ISSUE TITLE
drop zone in custom component in angular-formio

### ISSUE THREAD

**Issue by Dheeraj-Bharti** [formio/angular#625]
Hi,
I want to create a drop zone in my custom component that i've registered using the angular-formio. 
For ex - I want to make a dropzone in the primeng-panel.
Can you please let me know whether that is even feasible or not in angular-formio. 

Thank you,
Dheeraj

**Comment by travist** [comment]
Yes, this is feasible since custom components can be created using the "Angular" way described @ https://github.com/formio/angular-formio/wiki/Custom-Components-with-Angular-Elements

**Comment by Dheeraj-Bharti** [comment]
I have successfully created and registered the custom component but I want to drag and drop the components inside my custom component. For that I'm unable to find anything. We can create the custom component by extending the nested component in formiojs but how to do it for custom component in angular formio that i'm not able to figure out.

**Comment by travist** [comment]
I am not sure how that is done with the Custom Components feature since we did not develop it, but maybe @merobal knows since he developed it?

**Comment by merobal** [comment]
The custom component itself is an Angular component (in this context), so feel free to look at any dropzone implementations for Angular, and use that there.

**Comment by Dheeraj-Bharti** [comment]
I tried cdk drag and drop but in that I'm not able to drop the components, also it won't give a nested json i mean components inside components as we gets in formio. 
I want to create a custom layout component as we create in formiojs by extending the Nested Component with the dropzone provided as we have in layout components.

**Comment by merobal** [comment]
Try other d&d packages and try to compose the JSON inside the ng component and emit via the `value`. (Or you can use an Angular service to save data and combine during API calls.)

**Comment by Dheeraj-Bharti** [comment]
I'm trying with ngx-dnd, but how to get the list of components to my custom components.
Do you have any POC application for the same that you would to share.

**Comment by merobal** [comment]
what's the use-case you want to achieve?

**Comment by Dheeraj-Bharti** [comment]
I've added primeng-tabview ( https://primefaces.org/primeng/showcase/#/tabview ) to my custom component. I want to drag and drop components inside this just like we have tabs in layout section.

**Comment by Dheeraj-Bharti** [comment]
Any updates on this?

**Comment by merobal** [comment]
I haven't yet discovered how you can combine both scopes. Imagine that you have an Angular component and you want to include a part of an external lib, in this case, formio>angular>formio.

This Custom Components solution is designed for form fields (form inputs) when you wish to embed Angular fields inside formio, e.g. if you want to deliver the same UI across your Angular apps but enjoy the WYSIWYG-style form builder too.

For tabs, I recommend using the built-in formio's one and apply custom css to match the design with primeng.

**Comment by Seonho08** [comment]
@Dheeraj-Bharti Have you find solution to this, I too need to have drop zone for my custom component

**Comment by daneformio** [comment]
Closing this thread as it is outdated. Please re-open if it is still relevant. Thank you for your contribution!

**Comment by ZenMasterJacob20011** [comment]
What are you referring to when you say the base angular field components? Also now that creating custom components with angular components is deprecated the recommended way to create custom components would be to use plain javascript and Formio.use() to bring the custom component into your application. You can find documentation for this here
https://help.form.io/faq/tutorials-and-workflows/custom-component

**Comment by janstadt** [comment]
> What are you referring to when you say the base angular field components? Also now that creating custom components with angular components is deprecated the recommended way to create custom components would be to use plain javascript and Formio.use() to bring the custom component into your application. You can find documentation for this here https://help.form.io/faq/tutorials-and-workflows/custom-component

I presume (maybe incorrectly) that there were per form field angular components; eg text-area, input, etc, that could potentially be extended somehow. The info about just using vanilla js is sufficient though. I just wanted to make sure that there was a way to create more complex input fields that would work appropriately with formio.

**Comment by ZenMasterJacob20011** [comment]
We do not have per form field angular components. The formio angular library only exposes a form directive used to create form components from a json schema. We do not have "form field angular components". It looks like it was tried with https://github.com/formio/angular/wiki/Custom-Components-with-Angular-Elements but never took off. It would be best to create new custom component in vanilla javascript
